question26/newsComponents/main.js

```diff
@@ -0,0 +1,155 @@
+    tabContents.appendChild(createImgElements(newsData[i]));
+  }
+  contentsWrapper.appendChild(fragmentContents);
+  return contentsWrapper;
+}
+
+//get titles data
+const createTitles = ({ articles }) => {
+  const fragmentTitles = document.createDocumentFragment();
+  for (let i = 0; i < articles.length; i++) {
+    const articleTitle = articles[i].title;
+    const titleList = document.createElement('li');
+    const articleLink = document.createElement('a');
+    articleLink.href = "#";
+    articleLink.textContent = articleTitle;
+
+    // Display new icon
+    const elapsedTime = getElapsedDays(articles[i].date);
+    const newArrivalPeriod = 3;
+    if (elapsedTime <= newArrivalPeriod) {
+      const newIcon = createElementWithClassName("span", "new_icon"); 
+      newIcon.textContent = 'new';
+      articleLink.appendChild(newIcon);      
+    }
+
+    //add comment icon and number
+    const commentsLength = articles[i].comments.length; 
+    if (commentsLength > 0) {
+      articleLink.appendChild(createCommentIcon(articles[i]));
+    }
+    
+    fragmentTitles.appendChild(titleList).appendChild(articleLink);
+  }
+  return fragmentTitles;
+}
+
+//Get img data 
+const createImgElements = (newsData) => {
+  const imgWrapper = createElementWithClassName("div", "imgWrapper");
+  const imgTag = document.createElement('img');
+  imgTag.src = newsData.img;
+  imgWrapper.appendChild(imgTag);
+  return imgWrapper;
+}
+
+// Display comment icons and numbers
+const createCommentIcon = (articlesData) => {
+  const commentLength = articlesData.comments.length;
+  const commentIcon = createElementWithClassName("span", "comment_icon");
+  const commentNum = createElementWithClassName("span", "comment_icon_num");
+  const commentIconImg = createElementWithClassName("img", "comment_icon_img");
+  commentNum.textContent = commentLength;
+  commentIconImg.src = './img/comment_icon.png';
+  commentIcon.appendChild(commentIconImg);
+  commentIcon.appendChild(commentNum);
+  return commentIcon; 
+}
+
+// Get the number of days elapsed
+const getElapsedDays = (postDateData) => {
+const postedDate = new Date(postDateData);
+const today = new Date();
+const millisecondsPerDay = 24*60*60*1000;
+const elapsedDays = (today - postedDate) / millisecondsPerDay;
+return Math.floor(elapsedDays);
+}
+
+//Tab switching function
+tabsGroup.addEventListener("click", (e) => {
+  const activeTab = document.getElementsByClassName("is-active-tab")[0];
+  const activeContent = document.getElementsByClassName('is-active-content')[0];
+
+  activeTab.classList.remove("is-active-tab");
+  e.target.classList.add("is-active-tab");
+  activeContent.classList.remove('is-active-content'); 
+  const contents = document.getElementsByClassName('contentsContainer');
+  contents[e.target.dataset.index].classList.add('is-active-content'); 
+});
+
+const btn = document.getElementById('js-logout-btn');
```

Comment:
IMO
Insufficient variable names.
I think you should specify what kind of button it is.

Reply:
You are right. How about the correction? Please confirm.

fixed commit : https://github.com/csakyo/homework/pull/40/commits/cb9bb0307a1e6587a9e1f7d876cbf2065fead2ee